### ISSUE TITLE
Remove the `.well-known` resolution rule

### ISSUE THREAD

**Issue by OR13** [w3c-ccg/did-method-web#49]
This issue is raised to discuss potential breaking changes to the resolution rules for did web.

I propose we remove the `.well-known` resolution rule, and instead handle resolution for naked origins as follows:

`did:web:example.com` - > `https://example.com/did.json`

This is just another case of the existing path based routing resolution rules...

A 302 redirect can be configured to preserve any existing did documents.

This would simplify the did to url conversion logic by 50% with no loss of the name space.

**Comment by gribneau** [comment]
A Well Known URI (RFC [8615](https://www.rfc-editor.org/rfc/rfc8615.html) ) for a domain-wide DID would enable a sort of DKIM capability via DID, which I think could have considerable value.

I see that `/.well-known/did.json` is not currently in the [registry](https://www.iana.org/assignments/well-known-uris/well-known-uris.xhtml#well-known-uris-1). 

Are we anticipating that the registry entry will not be forthcoming?

**Comment by OR13** [comment]
@gribneau certainly we would need to register it if we continue to use it. 

I don't think we loose any potential features by opting not to use `.well-known` URIs though... there are similar situations today, for example https://github.com/OR13.gpg

**Comment by gribneau** [comment]
The usage of `.well-known` predates my involvement. I suppose I assumed that it was already registered or that registration was imminent.

If no registration is underway, we should not use it.

**Comment by dmitrizagidulin** [comment]
:+1: 
I'm on board with simplifying the "domain-only" resolution rules to drop the `.well-known/` part.

**Comment by mkhraisha** [comment]
How would this interact with routes that currently serve other .json at the base route?

assume example.com currently serves an json based api spec for example, how would we let that route know we're looking for the DID document?

**Comment by dmitrizagidulin** [comment]
Addressed by PR #50

**Comment by quartzjer** [comment]
I agree with [this comment](https://github.com/w3c-ccg/did-method-web/pull/50#pullrequestreview-834762306) by @tplooker and would 👎 dropping `.well-known/`.

The did:web ID is machine-translated into a HTTPS URI for automated GETs, it's not intended to be a directly fetch-able URI itself (IMO).
